projects/igniteui-angular/src/lib/data-operations/expressions-tree-util.ts

```diff
@@ -54,3 +58,159 @@ export class ExpressionsTreeUtil {
+        return DateTimeUtil.parseIsoDate(searchValue) ?? searchValue;
+    }
+
+    return searchValue;
+}
+
+/**
+ * Returns the filtering logic function for a given dataType and condition (contains, greaterThan, etc.)
+ * @param dataType The data type of the field.
+ * @param name The name of the filtering condition.
+ * @returns The filtering logic function.
+ */
+function getFilteringCondition(dataType: string, name: string): IFilteringOperation {
+    let filters: IgxFilteringOperand;
+    switch (dataType) {
+        case GridColumnDataType.Boolean:
+            filters = IgxBooleanFilteringOperand.instance();
+            break;
+        case GridColumnDataType.Number:
+        case GridColumnDataType.Currency:
+        case GridColumnDataType.Percent:
+            filters = IgxNumberFilteringOperand.instance();
+            break;
+        case GridColumnDataType.Date:
+            filters = IgxDateFilteringOperand.instance();
+            break;
+        case GridColumnDataType.Time:
+            filters = IgxTimeFilteringOperand.instance();
+            break;
+        case GridColumnDataType.DateTime:
+            filters = IgxDateTimeFilteringOperand.instance();
+            break;
+        case GridColumnDataType.String:
+        default:
+            filters = IgxStringFilteringOperand.instance();
+            break;
+    }
+    return filters.condition(name);
+}
+
+/**
+ * Recreates the IFilteringOperation for a given expression.
+ * If the `logic` is already populated - it will return the original IFilteringOperation
+ * of the expression.
+ * @param expression The expression for which to resolve the IFilteringOperation.
+ * @param dataType The data type of the field.
+ * @returns The IFilteringOperation for the given expression.
+ */
+function recreateOperatorFromDataType(expression: IFilteringExpression, dataType: string): IFilteringOperation {
+    if (!expression.condition?.logic) {
+        return getFilteringCondition(dataType, expression.conditionName || expression.condition?.name);
+    }
+
+    return expression.condition;
+}
+
+/**
+ * Recreates an expression from the given fields by applying the correct operands
+ * and adjusting the search value to be the correct type.
+ * @param expression The expression to recreate.
+ * @param fields An array of fields to use for recreating the expression.
+ * @returns The recreated expression.
+ */
+function recreateExpression(expression: IFilteringExpression, fields: FieldType[]): IFilteringExpression {
+    const field = fields?.find(f => f.field === expression.fieldName);
+
+    if (field && !expression.condition?.logic) {
+        if (!field.filters) {
+            expression.condition = recreateOperatorFromDataType(expression, field.dataType);
+        } else {
+            expression.condition = field.filters.condition(expression.conditionName || expression.condition?.name);
+        }
+    }
```

Comment:
in case the wrong conditionName is set on the column
```suggestion
        if (!expression.condition) {
            throw Error('Wrong `conditionName`, TBD message');
        }
```